src/IO/ADIOS/ADIOS1IOHandler.cpp

```diff
@@ -37,31 +95,527 @@ ADIOS1IOHandler::flush()
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_UCHAR:
+        {
+            auto ptr = reinterpret_cast< unsigned char* >(values.get());
+            auto const& vec = att.get< std::vector< unsigned char > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_UINT16:
+        {
+            auto ptr = reinterpret_cast< uint16_t* >(values.get());
+            auto const& vec = att.get< std::vector< uint16_t > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_UINT32:
+        {
+            auto ptr = reinterpret_cast< uint32_t* >(values.get());
+            auto const& vec = att.get< std::vector< uint32_t > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_UINT64:
+        {
+            auto ptr = reinterpret_cast< uint64_t* >(values.get());
+            auto const& vec = att.get< std::vector< uint64_t > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_FLOAT:
+        {
+            auto ptr = reinterpret_cast< float* >(values.get());
+            auto const& vec = att.get< std::vector< float > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_DOUBLE:
+        {
+            auto ptr = reinterpret_cast< double* >(values.get());
+            auto const& vec = att.get< std::vector< double > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_LONG_DOUBLE:
+        {
+            auto ptr = reinterpret_cast< long double* >(values.get());
+            auto const& vec = att.get< std::vector< long double > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+                ptr[i] = vec[i];
+            break;
+        }
+        case DT::VEC_STRING:
+        {
+            auto ptr = reinterpret_cast< char** >(values.get());
+            auto const& vec = att.get< std::vector< std::string > >();
+            for( size_t i = 0; i < vec.size(); ++i )
+            {
+                size_t size = vec[i].size() + 1;
+                ptr[i] = new char[size];
+                strncpy(ptr[i], vec[i].c_str(), size);
+            }
+            break;
+        }
+        case DT::ARR_DBL_7:
+        {
+            auto ptr = reinterpret_cast< double* >(values.get());
+            auto const& arr = att.get< std::array< double, 7> >();
+            for( size_t i = 0; i < 7; ++i )
+                ptr[i] = arr[i];
+            break;
+        }
+        case DT::BOOL:
+            throw std::runtime_error("No workaround for ADIOS1 bool implemented");
```

Comment:
Current PIConGPU representation: https://github.com/ComputationalRadiationPhysics/picongpu/pull/1756

also just as ints, unfortunately.

Reply:
Currently tinkering with this. Adapting the behaviour of using unsigned bytes for bools means we can store them but not read them back typesafe.